app/src/org/commcare/location/LocationRequestFailureHandler.kt

```diff
@@ -0,0 +1,27 @@
+package org.commcare.location
+import com.google.android.gms.common.api.ResolvableApiException
+
+object LocationRequestFailureHandler {
+
+    interface LocationResolutionCallback {
+        fun onResolvableException(exception: ResolvableApiException)
+        fun onNonResolvableFailure()
+    }
+
+    fun handleFailure(
+            failure: CommCareLocationListener.Failure,
+            callback: LocationResolutionCallback
+    ) {
+        when (failure) {
+            is CommCareLocationListener.Failure.ApiException -> {
+                val exception = failure.exception
+                if (exception is ResolvableApiException) {
+                    callback.onResolvableException(exception)
```

Comment:
we should instead call `((ResolvableApiException)exception).startResolutionForResult(this, LOCATION_SETTING_REQ);` here (look at `GeoPointActivity`)  directly ?